### ISSUE TITLE
InputBase Support

### ISSUE THREAD

**Issue by scottkuhl** [TimeWarpEngineering/blazor-state#232]
I need to capture state changes in an InputBase component for custom controls.  I can't inherit from BlazorStateComponent.

What's the best way to achieve this?

**Comment by ShadyNagy** [comment]
@scottkuhl 
We need more description of what you need with a small sample in github.

**Comment by scottkuhl** [comment]
Let's say I had a feature called ThemeState where you could set a value to Light, Dark or High Contrast.  I change the setting in a component and fire off the update:

`await Mediator.Send(new ThemeState.SetAction { Theme = "Light" });`

Now I have a component called MyCommandBar that inherits from ComponentBase that catches the change:

```
private ThemeState ThemeState => GetState<ThemeState>();
private string Css => Logic to set the CSS based on the current theme.
```

I change MyCommandBar from ComponentBase to BlazorStateComponent and everything works.  Amazing!

But now let's say I have a control called MyTextBox that inherits from InputBase<string> that I want to do virtually the same thing:

```
private ThemeState ThemeState => GetState<ThemeState>();
private string Css => Logic to set the CSS based on the current theme.
```

I can't change MyTextBox from InputBase<string> to ComponentBase so I can't change it to BlazorStateComponent.

A quick look through the source for InputBase is that in inherits from ComponentBase, so my first thought was to create a class BlazorStateInputBase that works similar to BlazorStateComponent.  But BlazorStateComponet is using static memory variables so it seems this might be more complicated to get all of this wired together.  

That's why I started this issue, to see if this is a problem that should be solved here by extending the library to support this scenario of custom controls.  Or is this a bad architecture pattern and InputBase should not listening for state changes.

**Comment by StevenTCramer** [comment]
Fixed with PR #263
